### PR TITLE
Add powerdns plugin

### PR DESCRIPTION
**Pull request by sileht**
Add powerdns plugin

### CODE REVIEW

**Comment by sileht** [comment]
@bastelfreak can you take a look ?

File: manifests/plugin/powerdns.pp [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# See http://collectd.org/documentation/manpages/collectd.conf.5.shtml#plugin_powerdns
+class collectd::plugin::powerdns (
+  $ensure                                 = 'present',
+  $order                                  = 10,
```
Comment (bastelfreak) [inline manifests/plugin/powerdns.pp]: please add datatypes for all parameters.

File: manifests/plugin/powerdns.pp [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  $interval                               = undef,
+  Optional[Hash[String, Hash]] $servers   = undef,
+  Optional[Hash[String, Hash]] $recursors = undef,
+  Optional[String] $local_socket          = undef,
+) {
+
+  include ::collectd
```
Comment (bastelfreak) [inline manifests/plugin/powerdns.pp]: please don't add the :: in front of classes

File: manifests/plugin/powerdns/recursor.pp [hunk 3]
```diff
@@ -0,0 +1,19 @@
+define collectd::plugin::powerdns::recursor (
+  $ensure    = 'present',
+  Optional[String] $socket = undef,
+  Array[String] $collect   = [],
```
Comment (bastelfreak) [inline manifests/plugin/powerdns/recursor.pp]: please align the = chars. For all the strings you could also do something like `Optional[String[1]]` which enforces a minimal string length of 1, so `''` isn't valid

File: manifests/plugin/powerdns/server.pp [hunk 4]
```diff
@@ -0,0 +1,20 @@
+define collectd::plugin::powerdns::server (
+  $ensure  = 'present',
+  Optional[String] $socket = undef,
+  Array[String] $collect   = [],
```
Comment (bastelfreak) [inline manifests/plugin/powerdns/server.pp]: please align the =

File: manifests/plugin/powerdns.pp (hunk 1 above)
Comment (sileht) [reply manifests/plugin/powerdns.pp]: done

File: manifests/plugin/powerdns.pp (hunk 2 above)
Comment (sileht) [reply manifests/plugin/powerdns.pp]: done

File: manifests/plugin/powerdns/recursor.pp (hunk 3 above)
Comment (sileht) [reply manifests/plugin/powerdns/recursor.pp]: done

File: manifests/plugin/powerdns/server.pp (hunk 4 above)
Comment (sileht) [reply manifests/plugin/powerdns/server.pp]: done

**Comment by bastelfreak** [comment]
thanks!

**Pull request merged**